shared/services/rewards/utils.go

```diff
@@ -310,46 +311,50 @@ func DownloadRewardsFile(cfg *config.RocketPoolConfig, interval uint64, cid stri
 			if err != nil {
 				return fmt.Errorf("error saving interval %d file to %s: %w", interval, rewardsTreePath, err)
 			}
-			return nil
-		}
-	}
 
-	return fmt.Errorf(errBuilder.String())
+			deserializedRewardsFile, err := DeserializeRewardsFile(writeBytes)
+			if err != nil {
+				return fmt.Errorf("Error deserializing file %s: %w", rewardsTreePath, err)
+			}
 
-}
+			// Get the original merkle root
+			downloadedRoot := deserializedRewardsFile.GetHeader().MerkleRoot
+			// Clear the merkle root so we have a safer comparison after calculating it again
+			deserializedRewardsFile.GetHeader().MerkleRoot = ""
 
-// Get CID for a serialized file
-func GetCIDForSerializedFile(data []byte, filename string) (cid.Cid, error) {
-	// Compress the data
-	encoder, _ := zstd.NewWriter(nil, zstd.WithEncoderLevel(zstd.SpeedBestCompression))
-	compressedData := encoder.EncodeAll(data, make([]byte, 0, len(data)))
+			// Reconstruct the merkle tree from the file data, this should overwrite the stored Merkle Root with a new one
+			deserializedRewardsFile.generateMerkleTree()
 
-	// Create an in-memory file and FS
-	mapFile := fstest.MapFile{
-		Data:    compressedData,
-		Mode:    0644,
-		ModTime: time.Now(),
-	}
-	fsMap := fstest.MapFS{filename: &mapFile}
-	file, err := fsMap.Open(filename)
-	if err != nil {
-		return cid.Cid{}, fmt.Errorf("error opening memory-mapped file: %w", err)
-	}
+			// Get the resulting merkle root
+			calculatedRoot := deserializedRewardsFile.GetHeader().MerkleRoot
 
-	// Use the web3.storage libraries to chunk the data and get the root CID
-	ds := dssync.MutexWrap(datastore.NewMapDatastore())
-	bsvc := bserv.New(blockstore.NewBlockstore(ds), nil)
-	dag := merkledag.NewDAGService(bsvc)
-	dagFmtr, err := adder.NewAdder(context.Background(), dag)
-	if err != nil {
-		return cid.Cid{}, fmt.Errorf("error creating DAG adder: %w", err)
-	}
-	root, err := dagFmtr.Add(file, "", fsMap)
-	if err != nil {
-		return cid.Cid{}, fmt.Errorf("error adding rewards file to DAG: %w", err)
+			// Compare the merkle roots to see if the original is correct
+			if downloadedRoot != calculatedRoot {
+				return fmt.Errorf("the merkle root from the downloaded file %s is not correct", rewardsTreePath)
+			}
+
+			if calculatedRoot != expectedRoot.String() {
```

Comment:
these are hex, so capitalization is actually kinda arbitrary, and i suggest using `strings.EqualFold` here and on L332

Reply:
Slightly out of scope but it would be nice to delete the `else` on L292 and outdent this whole block.

```golang
if thing {
    foo()
    continue;
} else {
    bar();
}
```
makes it less obvious that bar is unconditional when !thing

prefer:
```golang
if thing {
    foo()
    continue;
}

bar();
```